### ISSUE TITLE
luajit: ./misc/utils.lua:28: attempt to index local 'file' (a nil value)

### ISSUE THREAD

**Issue by jamiechoi1995** [doubledaibo/clcaption_nips2017#2]
When several epochs' training finished and the code start to save the checkpoint, the error message shows:

/home/wcai/torch/install/bin/luajit: ./misc/utils.lua:28: attempt to index local 'file' (a nil value)
stack traceback:
        ./misc/utils.lua:28: in function 'write_json'
        ./misc/net_utils.lua:398: in function 'language_eval'
        test.lua:261: in function 'evaluate_split'
        test.lua:405: in main chunk
        [C]: in function 'dofile'
        ...wcai/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x00405d50

But when I use the code of neuraltalk2, I can successfully save the checkpoint:

evaluating validation performance... 3200/3200 (9.179654)                           
validation loss:        9.1724067846911                                            
wrote json checkpoint to model_id.json                                              
iter 1: 9.124193                                                                    
iter 2: 9.075657 
...

**Comment by jamiechoi1995** [comment]
This issue is caused due to missing the folder coco-caption/val/

Just clone https://github.com/tylin/coco-caption.
